### PR TITLE
Support all float dtypes in `F.spatial_transformer_sampler`

### PR DESCRIPTION
**Pull request by takagi**
This PR follows #4582, generalize `F.spatial_transformer_sampler` to support all float dtypes.

### CODE REVIEW

**Comment by takagi** [comment]
Jenkins, test this please.

File: chainer/functions/array/spatial_transformer_sampler.py [hunk 1]
```diff
@@ -131,16 +132,17 @@ def backward_gpu(self, inputs, grad_outputs):
         grid_t = cuda.cupy.ascontiguousarray(grid_t)
         x = cuda.cupy.ascontiguousarray(x)
         gy = cuda.cupy.ascontiguousarray(gy)
-        gx = cuda.cupy.empty_like(x)
-        ggrid_t = cuda.cupy.empty_like(grid_t)
+        gx = cuda.cupy.zeros_like(x)
+        ggrid_t = cuda.cupy.zeros_like(grid_t)
```
Comment (toslunar) [inline chainer/functions/array/spatial_transformer_sampler.py]: Could you explain the change?

Comment (takagi) [reply chainer/functions/array/spatial_transformer_sampler.py]: I just left temporary change to find a reason of corrupting data. I will discard the change shortly.

Comment (toslunar) [reply chainer/functions/array/spatial_transformer_sampler.py]: Thanks.

**Comment by takagi** [comment]
Jenkins, test this please.

**Comment by takagi** [comment]
Jenkins, test this please.

**Pull request merged**